### ISSUE TITLE
activeClassName not working for hash fragments

### ISSUE THREAD

**Issue by entozoon** [rafgraph/react-router-hash-link#29]
Active class doesn't kick in if your link item scrolls to a hash fragment (whether on current page or not).
I.e. with an example such as this, you won't get the 'active' class at all, nor an `aria-current` value for that matter as mentioned in [issue #28](https://github.com/rafrex/react-router-hash-link/issues/28) (but this is not a dupe).
```jsx
<NavHashLink
    activeClassName="active"
    smooth
    to="/#anchor"
>
    Anchor
</NavHashLink>
```

**Comment by iamacatperson** [comment]
Any solution to this? I'm also facing the same problem.

**Comment by andrewdildy** [comment]
By default react-router matches your route against the pathname without the hash in order to determine whether the link is active.  

To override this you can pass a location parameter that reflects the current url
```jsx
<NavHashLink
    to="/#anchor"
    activeClassName="active"
    location={{pathname: document.location.pathname + document.location.hash}}
>
    Anchor
</NavHashLink>
```
or write a custom isActive function ([docs](https://reacttraining.com/react-router/web/api/NavLink/isactive-func)).

Pull request #20 also fixes this issue.

**Comment by rafrex** [comment]
I agree `<NavHashLink>` should only be active when both `path` and `hash` match. There are a number of edge cases around path matching that RR handles (trailing slash, when `to` is a function, etc). There is currently RR bug for determining when a link is active https://github.com/ReactTraining/react-router/issues/6877, when that is fixed I will add an additional check to this lib to make sure the hash's also match.

**Comment by klingerd** [comment]
I have the latest react-router and react-router-hash-link, and now andrewdidly's solution no longer seems to work.  I think react router is stripping off the hash from the "to" when determing isActive, and I don't get the data in the match parameter of the isActive() function either.  Is NavHashLink useless now, or am I just doing something wrong?

**Comment by simboonlong** [comment]
Workaround:

```
import { NavHashLink } from "react-router-hash-link";
import {
  useLocation
} from "react-router-dom";

...

export const YourComponent = () => {
  let location = useLocation();

  return (
    <NavHashLink
      to={item.path}
      className={`${location.pathname}${location.hash}` === item.path ? css["active"] : ""}
    >
        {item.name}
    </NavHashLink>
  );
};
```

**Comment by FinnGu** [comment]
> 
> 
> Workaround:
> 
> ```
> import { NavHashLink } from "react-router-hash-link";
> import {
>   useLocation
> } from "react-router-dom";
> 
> export const YourComponent = () => {
>   let location = useLocation();
> 
>   return (
>     ...
> 
>     <NavHashLink
>       to={item.path}
>       className={`${location.pathname}${location.hash}` === item.path ? "active" : ""}
>     >
>         {item.name}
>     </NavHashLink>
> 
>     ...
>   );
> };
> ```
> 
> item refers to nav items, you can output them in a loop.

In addition to the above, you should also set `activeClassName=""`. Otherwise the base Component will still set its className.

Here is an example if you are just interested in the hash and don't care about the path:
```
const CustomNavLink = ({ to, className, ...rest }: any) => {
    const location = useLocation();

    return (
        <NavLink
            to={to}
            className={`${className}${location.hash === to ? " active" : ""}`}
            activeClassName=""
            {...rest}
        />
    );
}
```

**Comment by GuidovdRiet** [comment]
Best way to fix this is to provide an [`isActive`](https://reactrouter.com/web/api/NavLink/isactive-func) property:

```
import { useLocation } from "react-router-dom";

const CustomNavLink = ({
  to,
  children
}) => {
  const location = useLocation();

  return (
    <NavLink
      to={to}
      activeClassName="active"
      isActive={() => to === location.pathname + location.hash}
    >
      {children}
    </NavLink>
  );
};

```

**Comment by rafgraph** [comment]
This should be fixed in v2.2.0

**Comment by JahnoelRondon** [comment]
None of these have worked for me, Not sure if it's ecause im using to= and then the exact name of the path like "#about" but they dont work unfortnately.
